pkg/operator/ceph/disruption/clusterdisruption/reconcile.go

```diff
@@ -267,18 +277,33 @@ func (c *ClusterMap) GetClusterNamespaces() []string {
 
 func (r *ReconcileClusterDisruption) deleteDrainCanaryPods(namespace string) error {
 	err := r.client.DeleteAllOf(context.TODO(), &appsv1.Deployment{}, client.InNamespace(namespace),
-		client.MatchingLabels{k8sutil.AppAttr: "rook-ceph-drain-canary"})
+		client.MatchingLabels{k8sutil.AppAttr: legacyDrainCanaryLabel})
 	if err != nil && !kerrors.IsNotFound(err) {
-		return errors.Wrap(err, "failed to delete all the legacy drain-canary pods")
+		return errors.Wrapf(err, "failed to delete all the legacy drain-canary pods with label %q", legacyDrainCanaryLabel)
 	}
 	return nil
 }
 
 func (r *ReconcileClusterDisruption) deleteLegacyPDBForOSD(namespace string) error {
 	err := r.client.DeleteAllOf(context.TODO(), &policyv1beta1.PodDisruptionBudget{}, client.InNamespace(namespace),
-		client.MatchingLabels{k8sutil.AppAttr: "rook-ceph-osd-pdb"})
+		client.MatchingLabels{k8sutil.AppAttr: legacyOSDPDBLabel})
 	if err != nil && !kerrors.IsNotFound(err) {
-		return errors.Wrap(err, "failed to delete all the legacy blocking podDisruptionBugets")
+		return errors.Wrapf(err, "failed to delete legacy OSD PDBs with label %q", legacyOSDPDBLabel)
 	}
 	return nil
 }
+
+func (r *ReconcileClusterDisruption) deleteLegacyPDBForMons(namespace string) error {
```

Comment:
What makes the legacy mon PDB actually "legacy"? What the difference between this one and the new one (if any)?

Reply:
The current mon PDB uses  `minAvailable= int32(float64((monCount + 1) / 2))`.  So for 3 mons, minAvaliable=2, for 5 mons minAvaliable=2 and likewise.  For this PR, we decided to play it safer and change it to maxUnavailable=1 (irrespective of the number of mons). 

In case of cluster upgrade, the mon pdb  with `minAvailable` should be the legacy pdb. We need to delete that pdb.

Reply:
@sp98 Isn't the legacy blocking PDB only for OSD pods? Are you going to change the PDB design then ?

$ oc get pdb -n openshift-storage
NAME                                              MIN AVAILABLE   MAX UNAVAILABLE   ALLOWED DISRUPTIONS   AGE
rook-ceph-mds-ocs-storagecluster-cephfilesystem   1               N/A               1                     30h
rook-ceph-mon-pdb                                 2               N/A               1                     30h
rook-ceph-osd                                     N/A             1                 1                     43m
[nberry@localhost logs]$

Reply:
we can discuss offline or on a call. @nehaberry

Reply:
I don't think we actually need to worry about a legacy pdb. The pattern I'm thinking is that when the cluster controller reconciles the mons, it will always update the mon pdb with the expected pdb. So the first time the operator runs with this change, the pdb would automatically be updated, and no need to delete the previous pdb.

Reply:
We can discuss more on this.  I would suggest having the `cluster disruption controller` to continue reconciling the mon pdbs, rather than moving it to `cluster controller`.

Reply:
Per our discussion, let's move the mon pdb reconcile code into the mon reconcile, and no need for it in the disruption reconciler. For example, around [line 300 of mon.go](https://github.com/rook/rook/blob/4d06ad1155e8eacdb44acb97cb46c88d98555716/pkg/operator/ceph/cluster/mon/mon.go#L300) after the mons are reconciled, we can insert a method that reconciles the mon pdb. The pdb would follow the pattern to create the pdb if it doesn't exist, and update it if it already exists (even if the spec didn't change). 

The `updateMonPDB()` method would just need to follow the pattern to generate the mon pdb spec and create the pdb if it doesn't exist yet, so some implementation will move over from the disruption controller.

Reply:
moved mon pdb reconcile code into mon reconcile. Testing looks good. This is ready for review now.